src/components/GoogleMapContainer.ts

```diff
@@ -216,16 +234,33 @@ class GoogleMapContainer extends Component<GoogleMapContainerProps, { alertMessa
         const locations = mxObjects.map(mxObject => {
             const lat = mxObject.get(this.props.latitudeAttribute);
             const lon = mxObject.get(this.props.longitudeAttribute);
+
             return {
                 address: mxObject.get(this.props.addressAttribute) as string,
                 latitude: lat ? Number(lat) : undefined,
-                longitude: lon ? Number(lon) : undefined
+                longitude: lon ? Number(lon) : undefined,
+                url: this.getMxObjectMarkerUrl(mxObject)
             };
         });
 
         this.setState({ locations });
     }
 
+    private getMxObjectMarkerUrl(mxObject: mendix.lib.MxObject): string {
+        const imageKey: string = mxObject.get(this.props.markerImageAttribute) as string;
+        const image = this.props.markerImages.find(value => value.enumKey === imageKey);
+
+        if (imageKey) {
```

Comment:
Should this be `if (imageKey && image) {` ??

Reply:
not necessary.
Implementation has changed to
```
  private getMxObjectMarkerUrl(mxObject: mendix.lib.MxObject): string {
        const imageKey: string = mxObject.get(this.props.markerImageAttribute) as string;
        const image = this.props.markerImages.find(value => value.enumKey === imageKey);

        return image
            ? getStaticMarkerUrl(image.enumImage as string, this.props.defaultMakerIcon)
            : "";
    }
```